### DISCUSSION TITLE
Translate Local Display coordinates to World

### DISCUSSION THREAD

**Discussion by kerim371** [Kitware/trame#606] [Q&A] [answered]
Hi,

Based on [Remote selection example](https://examples.vtk.org/site/Trame/Applications/RemoteSelection/RemoteSelection/) I'm trying to translate most likely `Local Display` coordinates to `World` (xyz of the scene). 

Doing it in the following form seems to bring incorrect results (I feel like the coordinates are shifted or something):
```python
# callback when selection is done
def on_box_selection_change(selection):

    x0 = selection["selection"][0]
    y0 = selection["selection"][2]
    x1 = selection["selection"][1]
    y1 = selection["selection"][3]

    coordinate = vtkCoordinate()
    coordinate.SetCoordinateSystemToDisplay()

    coordinate.SetValue(x0,y0)
    p0 = coordinate.GetComputedWorldValue(renderer)
    coordinate.SetValue(x1,y0)
    p1 = coordinate.GetComputedWorldValue(renderer)
    coordinate.SetValue(x1,y1)
    p2 = coordinate.GetComputedWorldValue(renderer)
    coordinate.SetValue(x0,y1)
    p3 = coordinate.GetComputedWorldValue(renderer)

    ...
```

I was hoping there is a way to make it work correct.

**Comment by jourdain** [comment, marked as answer]
Maybe [that is a good example for you](https://github.com/Kitware/trame-vtk/blob/master/examples/validation/PickingRemoteView.py).

**Reply by kerim371** [reply]
Yes , that was 100% what I was looking for! Thank you!
